server/game/cards/22-BtB/Pyke.js

```diff
@@ -0,0 +1,39 @@
+const DrawCard = require('../../drawcard.js');
+
+class Pyke extends DrawCard {
+    setupCardAbilities(ability) {
+        this.persistentEffect({
+            targetLocation: 'hand',
+            match: card => card.hasTrait('Warship'),
+            effect: ability.effects.gainAmbush(-1)
```

Comment:
Not sure if this is necessary, especially when considering it might not always be reduce (eg. if there is an ability that gives a card Ambush(X), X being its cost +1, rather than -1.